src/example-types/facet.js

```diff
@@ -134,5 +138,73 @@ module.exports = {
           }),
         options
       ),
-    })),
+    }))
+
+    let missedValues = _.difference(
+      values,
+      _.map(
+        ({ name }) => (node.isMongoId ? _.toString(name) : name),
+        results.options
+      )
+    )
+
+    let getMissedValues = (node, missedValues) =>
+      node.isMongoId ? _.map(ObjectID, missedValues) : missedValues
+
+    if (!_.isEmpty(missedValues)) {
+      let MissedResult = await search(
+        _.compact([
+          {
+            $match: {
+              [node.field]: { $in: getMissedValues(node, missedValues) },
+            },
+          },
+          { $group: { _id: `$${node.field}`, count: { $sum: 1 } } },
+          ...sortAndLimitIfNotSearching(node.optionsFilter, node.size),
+          ...lookupLabel(node),
+          _.get('label.fields', node) && projectStageFromLabelFields(node),
+          mapKeywordFilters(node),
+        ])
+      )
+      // when the value has been selected but filtered by query,the search result will still not contain the checked value. we return it with count 0
+      let missedByFilterValues = _.difference(
+        //when values are numeric values, stringify missedValues to avoid the bug.
+        _.map(_.toString, missedValues),
+        _.map(x => _.toString(x[`${node.field}`]), MissedResult)
+      )
+      let missedByFilterResult = []
+
+      if (!_.isEmpty(missedByFilterValues)) {
+        //use config to run runSearch(options, node, schema, filters, aggs)  function
+        missedByFilterResult = await config.getProvider(node).runSearch(
+          config.options,
+          node,
+          config.getSchema(node.schema),
+          {
+            [node.field]: { $in: getMissedValues(node, missedByFilterValues) },
+          },
+          [
+            { $group: { _id: `$${node.field}` } },
+            ...lookupLabel(node),
+            _.get('label.fields', node) && projectStageFromLabelFields(node),
+          ]
+        )
+      }
+
+      let missedValuesOptions = _.map(
+        ({ _id, label, count }) => ({
+          name: _id,
+          count,
+          ...facetValueLabel(node, label),
+        }),
+        _.flow(
+          _.map(({ _id, label }) => ({ _id, label, count: 0 })),
+          _.concat(MissedResult)
```

Comment:
I think it might be simpler if you just concat in one step at the end instead of here and also below - better separation of concerns that way

Reply:
I moved the _.map(({ _id, label }) => ({ _id, label, count: 0 })), to the previous function.